lib/alternatives.js

```diff
@@ -43,6 +43,9 @@ internals.Alternatives = class extends Any {
                         return baseType._validate(value, state, options);
                     }
                 }
+                else if (item.then === undefined && i < (il - 1)) {
```

Comment:
I think this needs a more important change. `baseType` should probably only ever be used as last resort, so removing it from the `failed` branch seem to make sense. I haven't tested it but my guts tell me we can simplify this code by only dealing with `baseType`  once.